libexec/python/base.py

```diff
@@ -126,6 +128,51 @@ def multi_package(func, args):
     return zip(itertools.repeat(func), args)
 
 
+def _strip_auth_presigned_s3(req, auth_query_params=['Signature',
+                                                     'X-Amz-Algorithm']):
+    """
+    Handles stripping authorization headers from requests to pre-signed S3
+    URLs. Without this we hit an InvalidArgument error about using more than
+    one auth mechanism.
+    """
+    if 'Authorization' not in req.headers:
+        return req
+
+    parts = urlparse(req.get_full_url())
+    host = parts.netloc
+    query_dict = parse_qs(parts.query)
+    is_s3_url = 's3' in host and host.endswith('.amazonaws.com')
```

Comment:
minor nit picking, but where s3 would appear, may be at least `'s3' in host.split('.')` would avoid matching some possibly unrelated urls (I don't know if any exist but I guess smth like `auths3.amazonaws.com` could emerge at some point)

Reply:
I do think that these edge cases are likely to occur (eventually) but we can address a more general strategy with subsequent PRs to this very specific use case - it would include more thinking about "all the possible versions of s3" and other storage providers generally, along with auth schemas and challenges. The way I'm thinking about it now is that this PR makes the software more functional than without, and is a step forward.

Reply:
@yarikoptic Unfortunately the URL varies with location/availability zone so we can't just look for `'s3' in host.split('.')`. Typically the URL starts with two sub-domains: the bucket name and a location-specific string starting with `s3`:

e.g. 

```
https://some-bucket-name.s3-us-west-1.amazonaws.com
```

So the current code looks for s3 anywhere in the string before the `.amazonaws.com` domain. The only update that would make this a bit more S3-specific without maintaining a list of availability zones in AWS is to instead check that first location-specific sub-domain for something that matches `^s3-.*$`. @vsoch Let me know if you'd prefer to have that update or leave things as-is.

Reply:
I think given that we can expect to later make this more elegant, it would suffice as is, but @yarikoptic 
 could you add your thoughts if this update would address some of your concern? There will always be "one more pattern" missed by a single regular expression, and when we have classes for each set it will be more manageable.

Reply:
Doing some research it appears there are several variations depending on whether you're using virtual-host or path style bucket URLS:

https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html#access-bucket-intro

And there's another one that's not listed in that doc according to this issue from another repo:

https://github.com/zalando/PGObserver/issues/91

I'll work on a regex that covers all those cases but there's still the possibility of other variants out there and the current code is pretty flexible as-is.